modules/ocf/manifests/ssl/lets_encrypt/dns.pp

```diff
@@ -10,24 +10,30 @@
     content => "${join($domains, ' ')} > ${title}",
   }
 
-  # Only run the dehydrated command to renew the cert if it is old enough (30
-  # days = 2592000 seconds left until it expires). dehydrated does this check
-  # itself, but this should help with puppet error spam a bit (for instance
-  # if Let's Encrypt is down for any reason) and mean that services can
-  # subscribe to this class to reload after new certs are obtained since this
-  # won't refresh unless dehydrated is re-run for some reason.
-  exec { "check ${title} cert expiration":
-    command => '/bin/true',
-    unless  => "openssl x509 -checkend 2592000 -noout -in /var/lib/lets-encrypt/certs/${title}/cert.pem",
-    user    => ocfletsencrypt,
-  } ~>
+  # Only run the dehydrated command to renew the cert if it is old enough, or
+  # missing some domains. dehydrated does the former check itself, but this
+  # should help with puppet error spam a bit (for instance if Let's Encrypt is
+  # down for any reason) and mean that services can subscribe to this class to
+  # reload after new certs are obtained since this won't refresh unless
+  # dehydrated is re-run for some reason.
+
+  $parsed_cert_info = parsejson($::le_cert_info)
+
+  $have_cert_info = $title in $parsed_cert_info
+  if $have_cert_info {
+    $cert_expires_soon = $parsed_cert_info[$title]['days_to_expiration'] < 30
+    # we subtract out any domains that are in the cert, and see if any are left
+    $cert_has_all_domains = ($domains - $parsed_cert_info[$title]['cert_names']) =~ Array[String, 0, 0]
```

Comment:
Nice, this is a pretty cool way of doing this. I wish there was a nicer syntax than `Array[String, 0, 0]` for this, but I don't see any way of doing this same kind of match.